### DISCUSSION TITLE
How to get click position from mouse event

### DISCUSSION THREAD

**Discussion by P1n3appl3** [wezterm/wezterm#5836] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

sway

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

Is there something like `pane:get_cursor_position()` that will return the position of the mouse cursor? Alternatively, is there some way to read the position of a click in the callback that it's bound to?

I wanted to bind ctrl+right click to open the clicked semantic zone in a pager or editor, which I've done before in kitty like this:
```
mouse_map ctrl+right press ungrabbed mouse_show_command_output
```

Initially I thought it'd be easy to script this behavior given the existing examples for [opening the whole scrollback in vim](https://wezfurlong.org/wezterm/config/lua/pane/get_lines_as_text.html#example-opening-whole-scrollback-in-vim) and [`get_zone_around_cursor()`](https://wezfurlong.org/wezterm/config/lua/pane/get_semantic_zone_at.html). Unfortunately I misunderstood and it seems that `pane:get_cursor_position()` gives the position of the text cursor not the mouse, so "get the region for the semantic zone around the position that the user just right clicked" isn't as simple as I'd hoped.

I looked into workarounds like making the click select a line of text at the (mouse) cursor and then somehow getting the position of the selection to grab the semantic zone around that, but I couldn't find a way to get the position of the selection either.

Am I missing something obvious here? If not is there a known workaround to handle click events when you want to take the location of the click into account?

**Comment by wez** [comment]
Take a look at https://wezfurlong.org/wezterm/config/lua/window/current_event.html
It doesn't really document the precise layout of the struct, but you can print/dump it and see what you get for the click event

**Reply by P1n3appl3** [reply]
I'd tried that but I didn't see a position in there. For example when I bind right click to this action:
```lua
action = wezterm.action_callback(function (window, _pane)
    print(window:current_event())
end)
```

this is what the table looks like

```
17:07:48.172  INFO   logging                         > lua: {
    "Down": {
        "button": "Right",
        "streak": 1,
    },
}
```
